src/main/java/uk/gov/pay/webhooks/validations/WebhookRequestValidator.java

```diff
@@ -16,16 +17,44 @@
 
 
 public class WebhookRequestValidator {
-    private static final Map<PatchPathOperation, Consumer<JsonPatchRequest>> patchOperationValidators = Map.of(
-            new PatchPathOperation(FIELD_DESCRIPTION, JsonPatchOp.REPLACE), JsonPatchRequestValidator::throwIfValueNotString,
-            new PatchPathOperation(FIELD_CALLBACK_URL, JsonPatchOp.REPLACE), RequestValidations::throwIfValueNotValidUrl,
-            new PatchPathOperation(FIELD_STATUS, JsonPatchOp.REPLACE), RequestValidations::throwIfValueNotValidStatusEnum,
-            new PatchPathOperation(FIELD_SUBSCRIPTIONS, JsonPatchOp.REPLACE), RequestValidations::throwIfValueNotValidSubscriptionsArray
-    );
-    private final JsonPatchRequestValidator patchRequestValidator = new JsonPatchRequestValidator(patchOperationValidators);
-
-    public void validateJsonPatch(JsonNode payload)  {
-        patchRequestValidator.validate(payload);
+    private final CallbackUrlService callbackUrlService;
+
+    private final JsonPatchRequestValidator liveContextValidator = validator(true);
+    private final JsonPatchRequestValidator testContextValidator = validator(false);
+
+    @Inject
+    public WebhookRequestValidator(CallbackUrlService callbackUrlService) {
+        this.callbackUrlService = callbackUrlService;
+    }
+
+    public void validate(JsonNode payload, Boolean isLiveContext)  {
```

Comment:
`isLiveContext` is a boolean acting as a poor man’s enum. I think I’d prefer an actual enum, particularly to make `WebhookRequestValidatorTest` nicer.

Reply:
👍 I'm in favour of this idea but not quite sure how it would work with the other boolean-y live checks throughout the app. I suppose this could set precedent for building the enum from a webhook entity and having it currently specify `TEST` and `LIVE` values. That could then be used by the validator and services further down